delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/client/ElasticSearchClient.scala

```diff
@@ -208,20 +209,24 @@ class ElasticSearchClient(client: HttpClient, endpoint: Uri, maxIndexPathLength:
     * @param refresh
     *   the value for the `refresh` Elasticsearch parameter
     */
-  def bulk(ops: Seq[ElasticSearchBulk], refresh: Refresh = Refresh.False): HttpResult[Unit] =
-    if (ops.isEmpty) IO.unit
+  def bulk(ops: Seq[ElasticSearchBulk], refresh: Refresh = Refresh.False): HttpResult[BulkResponse] = {
+    if (ops.isEmpty) IO.pure(BulkResponse.NoErrors)
     else {
       val bulkEndpoint = (endpoint / bulkPath).withQuery(Query(refreshParam -> refresh.value))
       val entity       = HttpEntity(`application/x-ndjson`, ops.map(_.payload).mkString("", newLine, newLine))
       val req          = Post(bulkEndpoint, entity).withHttpCredentials
+
       client
         .toJson(req)
-        .flatMap { json =>
-          IO.unless(json.hcursor.get[Boolean]("errors").contains(false))(
-            IO.raiseError(HttpClientStatusError(req, BadRequest, json.noSpaces))
-          )
+        .map {
+          case BatchPartialFailure(err) => err
+          case _                        => NoErrors
+        }
+        .recover { case BatchPartialFailure(err) =>
+          err
```

Comment:
It's unclear from the API documentation whether we get 200 or 400 in this situation, so we need to try and extract the partial error in both cases

Reply:
Did you get 400 answers for with mixed cases?

Reply:
I did once during testing.. 😒